forms/gridfield/GridFieldDetailForm.php

```diff
@@ -95,7 +95,7 @@ public function handleItem($gridField, $request) {
 
 		// if no validator has been set on the GridField and the record has a
 		// CMS validator, use that.
-		if(!$this->getValidator() && method_exists($record, 'getCMSValidator')) {
+		if(!$this->getValidator() && $record->hasMethod('getCMSValidator')) {
```

Comment:
Yep, you are right, that is my fear. Would there be any objection to the approach of abstracting to a simple function? That way any `instanceof Object` could be easily checked using the proper technique and then everything else would be checked via  `method_exists()`. 


> On Feb 28, 2015, at 9:54 AM, Daniel Hensby <notifications@github.com> wrote:
> 
> In forms/gridfield/GridFieldDetailForm.php:
> 
> > @@ -95,7 +95,7 @@ public function handleItem($gridField, $request) {
> >  
> >  		// if no validator has been set on the GridField and the record has a
> >  		// CMS validator, use that.
> > -		if(!$this->getValidator() && method_exists($record, 'getCMSValidator')) {
> > +		if(!$this->getValidator() && $record->hasMethod('getCMSValidator')) {
> I'm not certain that $record is always an Object here :/
> 
> —
> Reply to this email directly or view it on GitHub.
>